plugins/user-authenticators/saml2/src/org/apache/cloudstack/api/command/SAML2LogoutAPIAuthenticatorCmd.java

```diff
@@ -134,18 +159,17 @@ public String authenticate(String command, Map<String, Object[]> params, HttpSes
         String nameId = (String) session.getAttribute(SAMLPluginConstants.SAML_NAMEID);
         if (idpMetadata == null || nameId == null || nameId.isEmpty()) {
             try {
-                resp.sendRedirect(SAML2AuthManager.SAMLCloudStackRedirectionUrl.value());
+                SAMLUtils.redirectToSAMLCloudStackRedirectionUrl(resp, req);
             } catch (IOException ignored) {
                 s_logger.info("[ignored] final redirected failed.", ignored);
             }
             return responseString;
         }
-        LogoutRequest logoutRequest = SAMLUtils.buildLogoutRequest(idpMetadata.getSloUrl(), _samlAuthManager.getSPMetadata().getEntityId(), nameId);
 
         try {
-            String redirectUrl = idpMetadata.getSloUrl() + "?SAMLRequest=" + SAMLUtils.encodeSAMLRequest(logoutRequest);
+            String redirectUrl = SAMLUtils.buildLogoutRequestUrl(nameId, _samlAuthManager.getSPMetadata(), idpMetadata, SAML2AuthManager.SAMLSignatureAlgorithm.value());
             resp.sendRedirect(redirectUrl);
-        } catch (MarshallingException | IOException e) {
+        } catch (IOException e) {
```

Comment:
Did you intentionally take out the catch for the MarshallingException here? I'm assuming that this exception is still a possibility from sendRedirect()?